go/vt/vttablet/tabletmanager/shard_sync.go

```diff
@@ -236,23 +240,30 @@ func (agent *ActionAgent) startShardSync() {
 }
 
 func (agent *ActionAgent) stopShardSync() {
-	if agent.shardSyncCancel != nil {
-		agent.shardSyncCancel()
-		agent.shardSyncCancel = nil
-		agent.shardSyncChan = nil
+	agent.mutex.Lock()
+	if agent._shardSyncCancel != nil {
+		agent.mutex.Unlock()
```

Comment:
Shouldn't we call `_shardSyncCancel()` while still holding the lock? Otherwise we are reading it (to call it) without the lock. It's just a Context cancel function, which doesn't block, so there shouldn't be any risk of deadlock.